pkg/docker/events.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright © 2022 SUSE LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package docker handles port binding events from docker events API
+package docker
+
+import (
+	"context"
+
+	"github.com/Masterminds/log-go"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/rancher-sandbox/rancher-desktop-agent/pkg/tracker"
+)
+
+const (
+	startEvent = "start"
+	stopEvent  = "stop"
+)
+
+// EventMonitor monitors the Docker engine's Event API
+// for container events.
+type EventMonitor struct {
+	dockerClient *client.Client
+}
+
+// NewEventMonitor creates and returns a new Docker API client.
+func NewEventMonitor() (*EventMonitor, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
```

Comment:
Will this need some sort of loop (because the docker daemon might not have finished starting up yet)?

Reply:
@mook-as I have a retry mechanism that checks for the existence of docker socket file `\\.\pipe\docker_engine` before it event triggers `NewEventMonitor`. 

The problem is that if the docker engine is not running it errors out at the later stage (when querying for the events) and not while initializing the client. This retry mechanism will be in the main you will see it once things are wired up.

Reply:
Now that I have the surrounding context this is fine (though it would be good to include the precondition in the function documentation).